### DISCUSSION TITLE
Will the package be ported to Vue 3 any time soon?

### DISCUSSION THREAD

**Discussion by smolinari** [Akryum/vue-apollo#1132] [General]
Would be great if it would be, since Vue 3 is out of beta. 

Scott

**Comment by Manubi** [comment]
It's possible to use right now, but I wouldn't recommend it. The documentation is not that good as you have to deal with errors in the example code and if you want to fix the documentation you get a 404. 

I am trying to use it with the composition API.

As the last release was 4 months ago I don't think it's high on their priority list...

**Comment by smolinari** [comment]
I've got a vue-apollo app-extension for Quasar Framework and am wanting to move it forward to also support Vue3. Quasar just came out with their v2 supporting Vue3. So I really wish this project would get some TLC from Guillaume. 😃 I'm worried the approach might change with the APIs. If I can at least get a "they won't change much", I can continue, I think.

Scott

**Reply by Akryum** [reply]
APIs will not change much from what is in v4 alpha

**Comment by Akryum** [comment]
Technically `@vue/apollo-composable` is already compatible with Vue 3, but it's using the `onServerPrefetch` hook which is not implemented on Vue 3 [yet](https://github.com/vuejs/vue-next/pull/3070), which can cause issues with people using either TypeScript, SSR or Vite.

**Reply by Manubi** [reply]
Hey,
the edit link to github is not working. 
https://v4.apollo.vuejs.org/guide-composable/pagination.html#the-connection-directive
I guess the cursor should be `result.value.feed`?

**Reply by smolinari** [reply]
@Akryum - Is the options API version of vue-apollo 4 also compatible to Vue 3? I'm not able to get it to work with `app.use`.

I'd like to offer both APIs within the Quasar extension, but I'm unable to get the options API to work. 

I'm also having an issue with `app.provide`. If I inject the apollo client with that, I get this error.

`Uncaught (in promise) Error: Apollo client with id default not found. Use provideApolloClient() if you are outside of a component setup.`

I've got a work around for providing (it's ugly though) and composition API with vue-apollo4 is working. Just wondering if you have any suggestions to tie in vue-apollo4 with Vue's created  `app` instance. 

Scott

**Reply by fkromer** [reply]
> 
> 
> Technically `@vue/apollo-composable` is already compatible with Vue 3, but it's using the `onServerPrefetch` hook which is not implemented on Vue 3 [yet](https://github.com/vuejs/vue-next/pull/3070), which can cause issues with people using either TypeScript, SSR or Vite.

An update for future readers: The PR has been merged into master.

**Comment by backbone87** [comment]
is there anything, where we could help to move this into beta? some features are not released on alpha yet like https://github.com/vuejs/vue-apollo/pull/1062 which is a pretty important one
